### PR TITLE
Separate steps in back up process

### PR DESCRIPTION
**Pull request by hoangdat**
#### Reset in the case of keys conflicted between Recovery Vault and Account Data
[Demo_reset_backUp.webm](https://github.com/linagora/twake-on-matrix/assets/6462404/e1388b48-b33c-4a44-89c4-bc3c8aa55c33)

#### Set up new key from scratch
https://github.com/linagora/twake-on-matrix/assets/6462404/81cad586-3bc5-4847-84ae-cc1a65277583

#### Login in the next time, load and unlock keys
https://github.com/linagora/twake-on-matrix/assets/6462404/27b55226-84e7-4442-9a8c-ada643bfb60e

### CODE REVIEW

File: lib/pages/bootstrap/tom_bootstrap_dialog.dart [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
+import 'package:fluffychat/domain/usecase/recovery/delete_recovery_words_interactor.dart';
+import 'package:fluffychat/domain/usecase/recovery/save_recovery_words_interactor.dart';
 import 'package:fluffychat/widgets/adaptive_flat_button.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/l10n.dart';
 import 'package:matrix/encryption.dart';
 import 'package:matrix/encryption/utils/bootstrap.dart';
 import 'package:matrix/matrix.dart';
 
+import 'bootstrap_dialog.dart';
```
Comment (nqhhdev) [inline lib/pages/bootstrap/tom_bootstrap_dialog.dart:12]: pls import relative path

File: lib/pages/bootstrap/tom_bootstrap_dialog.dart [hunk 2]
```diff
@@ -32,8 +38,13 @@ class TomBootstrapDialog extends StatefulWidget {
...
   final _saveRecoveryWordsInteractor = getIt.get<SaveRecoveryWordsInteractor>();
+  final _deleteRecoveryWordsInteractor =
+      getIt.get<DeleteRecoveryWordsInteractor>();
   Bootstrap? bootstrap;
+
   String? titleText;
+  Widget? body;
+  final buttons = <AdaptiveFlatButton>[];
 
   UploadRecoveryKeyState _uploadRecoveryKeyState =
```
Comment (nqhhdev) [inline lib/pages/bootstrap/tom_bootstrap_dialog.dart:49]: Why don't you change `ValueNotifier<UploadRecoveryKeyState>`

Comment (hoangdat) [reply lib/pages/bootstrap/tom_bootstrap_dialog.dart:49]: use `setState` instead

File: lib/pages/bootstrap/tom_bootstrap_dialog.dart (hunk 1 above)
Comment (hoangdat) [reply lib/pages/bootstrap/tom_bootstrap_dialog.dart:12]: but it stay in the same folder

File: lib/data/network/recovery_words/recovery_words_api.dart [hunk 3]
```diff
@@ -26,4 +28,21 @@ class RecoveryWordsAPI {
         .onError((error, stackTrace) => throw Exception(error));
     return response.statusCode >= 200 && response.statusCode <= 299;
   }
+
+  Future<bool> deleteRecoveryWords() async {
+    final options = Options(
+      validateStatus: (status) {
+        if (status == null) return false;
+        return status >= 200 && status <= 299 || status == 404;
```
Comment (sherlockvn) [inline lib/data/network/recovery_words/recovery_words_api.dart:36]: please make a function for it

File: lib/data/network/recovery_words/recovery_words_api.dart [hunk 4]
```diff
@@ -26,4 +28,21 @@ class RecoveryWordsAPI {
...
+      },
+    );
+    final response = await _client
+        .delete(TomEndpoint.recoveryWordsServicePath.path, options: options)
+        .onError((error, stackTrace) {
+      Logs().e('RecoveryWordsAPI::deleteRecoveryWords() [Exception]', error);
+      throw Exception(error);
+    });
+    return response.statusCode >= 200 && response.statusCode <= 299 ||
```
Comment (sherlockvn) [inline lib/data/network/recovery_words/recovery_words_api.dart:45]: same

**Comment by hoangdat** [comment]
work well on tom-dev but not in linagora. Debuging more

**Comment by hoangdat** [comment]
### back up run after init
[waiting_for_complete_init.webm](https://github.com/linagora/twake-on-matrix/assets/6462404/e3288113-20eb-405d-a14a-3b37922ccfb1)

**Review by nqhhdev — APPROVED** [review]

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**